### ISSUE TITLE
Annotation to suppress miniboxing warnings

### ISSUE THREAD

**Issue by VladUreche** [miniboxing/miniboxing-plugin#194]
Great idea! Thanks to @julien-truffaut for coming up with this. Now let's see what a good implementation of the idea would be:
- an annotation like `@suppresWarnings("unchecked")` would be too coarse-grained
- an annotation like `@generic` for type parameters would be better -- it could be used to prevent warnings from both the definition-site or the use-site, but it may become verbose at some point -- but what would be a good name for this?
### Use-site:

``` scala
scala> def foo[@miniboxed T](t: T) = t
defined method foo
```

Before:

``` scala
scala> foo[Any](3)
<console>:9: warning: Using the type argument "Any" for the miniboxed type parameter T of method foo is not specific enough, as it could mean either a primitive or a reference type. Although method foo is miniboxed, it won't benefit from specialization:
              foo[Any](3)
                 ^
res0: Any = 3
```

After:

``` scala
scala> foo[Any @generic](3) = 3
res1: Any = 3
```
### Definition-site:

Before:

``` scala
scala> def foo[T](t: T) = t
defined method foo

scala> foo[Int](3)
<console>:9: warning: The method foo would benefit from miniboxing type parameter T, since it is instantiated by a primitive type.
              foo[Int](3)
                 ^
res2: Int = 3
```

After:

``` scala
scala> def foo[@generic T](t: T) = t
defined method foo

scala> foo[Int](3) = 3 // no warning!
res3: Int = 3
```

WDYT? What's a good name for this? And would you like to have the `@suppressWarnings`-like method/class-level annotation?

**Comment by VladUreche** [comment]
Related: #183 -- suppress warnings from implicitly inferred arguments.

**Comment by VladUreche** [comment]
Closed thanks to @milosstojanovic's PR #197.
